### PR TITLE
[cli] add cli support for region code

### PR DESCRIPTION
**Pull request by gjc13**
This allows users to get/set the region code over OpenThread cli.

### CODE REVIEW

File: src/agent/ncp_openthread.cpp [hunk 1]
```diff
@@ -60,6 +60,11 @@ using std::chrono::steady_clock;
 namespace otbr {
 namespace Ncp {
 
+otCliCommand ControllerOpenThread::sRegionCommand = {
```
Comment (bukepo) [inline src/agent/ncp_openthread.cpp:63]: constexpr?

**Review by bukepo — APPROVED** [review]
LGTM:+1:

File: src/agent/ncp_openthread.cpp (hunk 1 above)
Comment (gjc13) [reply src/agent/ncp_openthread.cpp]: `const` will be more proper for these callback structs.

File: src/agent/ncp_openthread.cpp [hunk 2]
```diff
@@ -322,6 +328,37 @@ void ControllerOpenThread::RegisterResetHandler(std::function<void(void)> aHandl
...
+    ControllerOpenThread *controller = static_cast<ControllerOpenThread *>(aContext);
+    controller->HandleRegionCommand(aArgLength, aArgs);
+}
+
+void ControllerOpenThread::HandleRegionCommand(uint8_t aArgLength, char **aArgs)
+{
+    if (aArgLength == 0)
+    {
+        otCliOutputFormat("%s\n", mRegionCode.c_str());
+        otCliOutputFormat("Done\n", mRegionCode.c_str());
```
Comment (bukepo) [inline src/agent/ncp_openthread.cpp:342]:
```suggestion
        otCliOutputFormat("Done\n");
```

File: src/agent/ncp_openthread.cpp [hunk 3]
```diff
@@ -322,6 +328,37 @@ void ControllerOpenThread::RegisterResetHandler(std::function<void(void)> aHandl
...
+    else if (aArgLength == 1)
+    {
+        if (strnlen(aArgs[0], 3) == 2)
+        {
+            mRegionCode = aArgs[0];
+            otCliOutputFormat("Done\n");
+        }
+        else
+        {
+            otCliOutputFormat("Error: InvalidArgs\n");
```
Comment (bukepo) [inline src/agent/ncp_openthread.cpp:353]: not in this PR, but prefer change the prototype of otCliCommand to return otError and handle error message in OpenThread. Just like the diagnostic commands.

**Review by jwhui — APPROVED** [review]

**Pull request merged**